packages/zoe/src/cleanProposal.js

```diff
@@ -162,10 +164,41 @@ export const cleanProposal = (proposal, getAssetKindByBrand) => {
   const cleanedProposal = harden({
     want: cleanedWant,
     give: cleanedGive,
+    multiples,
     exit,
   });
   fit(cleanedProposal, ProposalShape);
+  if (multiples > 1n) {
+    for (const amount of values(cleanedGive)) {
+      assert.typeof(
+        amount.value,
+        'bigint',
+        X`multiples > 1 not yet implemented for non-fungibles: ${multiples} * ${amount}`,
+      );
+    }
+  }
   assertExit(exit);
   assertKeywordNotInBoth(cleanedWant, cleanedGive);
   return cleanedProposal;
 };
+
+/**
+ *
+ * @param {Amount} amount
+ * @param {bigint} multiples
+ * @returns {Amount}
+ */
+export const scaleAmount = (amount, multiples) => {
+  if (multiples === 1n) {
+    return amount;
+  }
+  const { brand, value } = amount;
+  assert(value >= 1n);
+  assert.typeof(
+    value,
+    'bigint',
+    X`multiples > 1 not yet implemented for non-fungibles: ${multiples} * ${amount}`,
+  );
+  return harden({ brand, value: value * multiples });
```

Comment:
don't we use natSafeMath even for addition and multiplication?

Reply:
Done. Also for the divide in offerSafety.js where it was even more valuable because it explicitly states it is doing a floorDivide, which is important for any reader to know.